### ISSUE TITLE
Allow chains of field accesses without having to specify each subchain separately

### ISSUE THREAD

**Issue by anthonyvdotbe** [uber/NullAway#1529]
Consider a deeply nested `@NonNull` object `foo` whose entire state is `@Nullable`.
All classes involved are as dumb as can be, just a bunch of `@Nullable` fields.
While converting this untrusted input object `foo` to a trusted domain object, I'd like to be able to write code like:

```java
Objects.requireNonNull(foo.deeply.nested.bar.baz);
```

instead of:

```java
Objects.requireNonNull(foo.deeply);
Objects.requireNonNull(foo.deeply.nested);
Objects.requireNonNull(foo.deeply.nested.bar);
Objects.requireNonNull(foo.deeply.nested.bar.baz);
```

with the rationale that something like:

```java
Objects.requireNonNull(foo.deeply.nested.bar.baz);
```

clearly communicates the intention of:

> if any of `foo.deeply`, `foo.deeply.nested`, etc. are null, I'm fine with a `NullPointerException` being thrown

**Comment by msridhar** [comment]
It's an interesting proposal.  I think what you're asking is for some kind of API `requireDeepNonNull` (naming TBD) which, for a call `requireDeepNonNull(e)`, NullAway treats as follows:

1. Treat `e` as non-null after the call
2. Treat any de-referenced sub-expression of `e` as non-null after the call, _and_ don't give any NullAway warnings for those de-referenced sub-expressions

I think this captures the behavior you want; do you agree?

**Comment by anthonyvdotbe** [comment]
> I think this captures the behavior you want; do you agree?

Yes, I fully agree.

**Comment by msridhar** [comment]
@anthonyvdotbe #1566 shows a trick for achieving what you want with a local `@SuppressWarnings` on a variable declaration.  Given that this seems to work, I'm not inclined to add further specialized support for it.  What do you think?

**Comment by anthonyvdotbe** [comment]
@msridhar that's a neat trick indeed. I agree further specialized support is unneeded at this point, so I'll go ahead and close this. Thanks for your help.
